internal/controller/controllers/agent_controller.go

```diff
@@ -299,7 +482,18 @@ func (r *AgentReconciler) updateStatus(ctx context.Context, log logrus.FieldLogg
 		}
 
 		if h.Progress != nil && h.Progress.CurrentStage != "" {
-			agent.Status.Progress.CurrentStage = h.Progress.CurrentStage
+			var isNoneHostRebooting bool
+
+			if isNoneHostRebooting, err = r.isDay2NonePlatformHostRebooting(ctx, agent, h); err != nil {
+				log.WithError(err).Errorf("Failed to find if agent %s/%s belongs to none platform cluster and is rebooting", agent.Namespace, agent.Name)
+				return ctrl.Result{RequeueAfter: defaultRequeueAfterOnError}, nil
+			}
+
+			if isNoneHostRebooting {
```

Comment:
what is difference between None and BM stage in rebooting?

Reply:
maybe better to have function like 
setCurrentStage and move all the logic there? it will be mostly same as isDay2NonePlatformHostRebooting ?